### ISSUE TITLE
Failed to complete #destroy action should be a warning, not an error

### ISSUE THREAD

**Issue by gdamiani** [test-kitchen/kitchen-ec2#191]
This should be a warning, not an error.  Sometimes our build system has reason to destroy an instance.  Sometimes test kitchen cleans it up.  In the case of the former, Test Kitchen throws this error and refuses to build the next iteration until we jump on the build host and clear out the `.kitchen` directory.  If the instance is terminated already, why do we care enough to throw an error?  Seems like a test EC2 instance is ephemeral anyway.  If it terminated unexpectedly, that was probably caught by the test iteration in which it occurred.  It should have no effect on the subsequent test iteration.

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #destroy action: [The instance ID 'i-b2b28c60' does not exist]
```

**Comment by gdamiani** [comment]
In our particular case, just running `kitchen destroy` isn't good enough to clean up after the previous test kitchen run:
```
jenkins@jenkops:~/jobs/cookbook/workspace/cookbooks/cookbook$ kitchen destroy
-----> Starting Kitchen (v1.4.2)
-----> Destroying <default-ubuntu-1204>...
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #destroy action: [The instance ID 'i-b2b2b2b2' does not exist]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by vt102** [comment]
Just ran into the same issue.

**Comment by rljohnsn** [comment]
My kitchen destroy has no affect. Instance continues to run even though kitchen list shows it as "<Not Created>"

**Comment by adimarco** [comment]
I'm seeing the same thing.  Removing the associated .yml file in .kitchen that contained the no-longer-existing instance ID is a valid workaround, but this should really be a warning.

**Comment by kzw** [comment]
I get this error usually but not always whenever I run with '-c number'  

It does not normally happen when I run without '-c'

**Comment by cheeseplus** [comment]
The problem here is that the behaviour is correct and is a core piece of what kitchen expects. The same goes for any driver so just because ec2 _feels_ ephemeral to some doesn't mean the same for others. From kitchen's POV this is an error: it created a machine, added it to the statefile, and when you try to destroy it via kitchen it has disappeared so it can't do what it's told - this constitutes an error in the common use case.

Changing the existing behaviour doesn't strike me as a good option as kitchen as these out of band destroys can happen just as unintentionally as not so the safest thing to do is error so that an operator can investigate. I've stated before in a handful of other threads that this is expected behaviour but what we could do is add an option to ignore that the instance does not exist. This wouldn't change the default but would only warn if invoked. If someone wants to write that up in a PR, we'd consider it.

**Comment by SydOps** [comment]
For example, I use terraform to build and deploy aws ec2/vpc/s3 and other resources, when I try to destroy it, some resources were stopped, terminated. But `terraform destroy` doesn't report error, and keep cleaning the rest until all related resources cleaned.  

I'd like to have same feature in kitchen test, or kitchen-ec2.

**Comment by gigawatts** [comment]
I would love to see this solved, and not just for the ec2 driver. Adding a +1 to have this bumped up in priority.

**Comment by thepatrick** [comment]
I disagree that this behavior is desirable. `kitchen destroy` should (in the EC2 case) destroy the instance, but if it's already destroyed then... well that's exactly what we wanted anyway, so how does an error make sense?

**Comment by bender-the-greatest** [comment]
A suggestion here - maybe `kitchen destroy` still operates the way it does, but add a `--force` parameter or `--update-if-missing` (pardon my poor parameter names) so that the end user can explicitly clean up the inconsistent state should it occur. This would allow cleanup of this state a simple command to run rather than having to delete a file within `.kitchen` when the case is hit.

**Comment by ozbillwang** [comment]
another use case. 

our support team usually kill some ec2 instances, if they think that's not with proper tags, or low CPUs, especially in non-prod aws accounts.  They have highest permission and manage their own step-funciton or lambda functions to monitor and terminate these un-used ec2 instances to save the cost. 

Because of that, I got the error some time. So it would be great if kitchen test marked that ec2 instance has been destroyed and continous rest tasks, not mark it as block error.

**Comment by clintoncwolfe** [comment]
I just tried this - creating an instance via TK 2.5.2, kitchen-ec2 3.7.1 - then going into the AWS console and terminating the instance - and then running `kitchen destroy` within a few minutes of termination. For me, no error occurred in the case that the instance still exists in AWS records; TK is able to destroy the instance. No surprise.

Next, I repeated those steps but waited until the instance was no longer listed in the AWS Console before running `destroy`. In that case, I got the helpful message:
```
$$$$$$ Received The instance ID 'i-...' does not exist, instance was probably already destroyed. Ignoring
Waited 0/300s for instance <i-0a2d23bfbdacf624d> to terminate.
```
and then a full 300 sec wait, after which it failed to destroy the instance. 

Correct or no, I feel like a better experience could be had.

**Comment by tas50** [comment]
It feels like a solid improvement we could make on the destroy is that if the instance does not come back from AWS, we assume it's already been destroyed. That way we don't spin for 300 seconds before we move on.
